### PR TITLE
Bug and Compiler fixes

### PR DESCRIPTION
**Pull request by srsampson**
Fix bug in phase.c and compiler changes to STM32

### CODE REVIEW

File: src/phase.c [hunk 1]
```diff
@@ -241,6 +241,13 @@ void mag_to_phase(float phase[],             /* Nfft/2+1 output phase samples in
     int  Ns = Nfft/2+1;
     int  i;
 
+    /* initialize array */
+
+    for(i=0; i<Nfft; i++) {
+        Sdb[i].real = 0.0;
+        Sdb[i].imag = 0.0;
+    }
+
```
Comment (drowe67) [inline src/phase.c]: Hi @srsampson - so why is this needed given the next few lines that fill ```Sdb[]```?  Do they miss some values or did the compiler complain?

File: stm32/src/si53xx.c [hunk 2]
```diff
@@ -42,6 +42,7 @@
 #include "math.h"
 #include "si53xx.h"
 
+T_Si5351_data   Si5351_Config;
```
Comment (drowe67) [inline stm32/src/si53xx.c]: As you suggested on the mailing list, I'm OK simply rm-ing this and other sm2000 specific code.  It will still be there in the Git basement if we ever want it.

**Comment by drowe67** [comment]
@srsampson I think the Travis fail is an intermittent LDPC test failure I've also been experiencing. It's probably a precision thing.  I will need to sort that out some time.

Also - for any stm32 PRs:
1.  We need to run and pass all of the stm32 unit tests
1.  And test the sm1000 binary on a SM1000, e.g. try decoding some canned files (70)D and 1600), and throw it out to sm100 users for testing.

File: src/phase.c (hunk 1 above)
Comment (srsampson) [reply src/phase.c]: I believe the next few lines only initialize 0 - Ns while the array is 0 - Nfft thus half the values I assume will be random stack values.

This seemed suspicious to me.

Comment (srsampson) [reply src/phase.c]: > I believe the next few lines only initialize 0 - Ns while the array is 0 - Nfft thus half the values I assume will be random stack values.
> 
> This seemed suspicious to me.

Oh darn, I have made a BIG mistake here. I completely missed the code inside the loop!

I feel maybe this whole PR is a wash. Probably needs deleted. argh!
